lib/taskana-core-test/src/test/java/acceptance/task/update/UpdateTaskAccTest.java

```diff
@@ -0,0 +1,561 @@
+
+    Instant planned = Instant.parse("2024-06-17T23:14:31.0Z");
+    task.setPlanned(planned);
+    Task updatedTask = taskService.updateTask(task);
+    updatedTask = taskService.getTask(updatedTask.getId());
+
+    assertThat(updatedTask.getPlanned()).isEqualTo(planned);
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_ThrowException_When_ModificationOfWorkbasketKeyIsAttempted() throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService, "admin");
+    WorkbasketSummary workbasketSummary =
+        defaultTestWorkbasket().buildAndStoreAsSummary(workbasketService, "admin");
+
+    ((TaskImpl) task).setWorkbasketKey(workbasketSummary.getKey());
+
+    assertThatThrownBy(() -> taskService.updateTask(task))
+        .isInstanceOf(InvalidArgumentException.class);
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_UpdateNoTasks_When_UpdateTasksWithUnmatchedObjectReferenceWasCalled()
+      throws Exception {
+    TaskBuilder.newTask()
+        .classificationSummary(defaultClassificationSummary)
+        .workbasketSummary(defaultWorkbasketSummary)
+        .primaryObjRef(defaultObjectReference)
+        .buildAndStore(taskService, "admin");
+
+    ObjectReferenceImpl por = new ObjectReferenceImpl();
+    por.setCompany("00");
+    por.setSystem("PASystem");
+    por.setSystemInstance("00");
+    por.setType("VNR");
+    por.setValue("22334455");
+    Map<TaskCustomField, String> customProperties = new HashMap<>();
+    customProperties.put(CUSTOM_7, "This is modifiedValue 7");
+    customProperties.put(CUSTOM_14, null);
+    customProperties.put(CUSTOM_3, "This is modifiedValue 3");
+    customProperties.put(CUSTOM_16, "This is modifiedValue 16");
+
+    List<String> taskIds = taskService.updateTasks(por, customProperties);
+    assertThat(taskIds).isEmpty();
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_UpdateTasks_When_MatchingPrimaryObjectReferenceWasChanged() throws Exception {
+    ObjectReference objectReference =
+        ObjectReferenceBuilder.newObjectReference()
+            .company("00")
+            .system("PASystem")
+            .systemInstance("00")
+            .type("VNR")
+            .value("22334455")
+            .build();
+    TaskBuilder.newTask()
+        .classificationSummary(defaultClassificationSummary)
+        .workbasketSummary(defaultWorkbasketSummary)
+        .primaryObjRef(objectReference)
+        .buildAndStore(taskService, "admin");
+    TaskBuilder.newTask()
+        .classificationSummary(defaultClassificationSummary)
+        .workbasketSummary(defaultWorkbasketSummary)
+        .primaryObjRef(objectReference)
+        .buildAndStore(taskService, "admin");
+
+    ObjectReferenceImpl por = new ObjectReferenceImpl();
```

Comment:
Why do you need this? It has the same properties as "objectReference". Could you just reuse "objectReference"?